File: SORTING/INSERTIONSORT/Insertionsort.js
function InsertionSort(arr){
    for( let i = 1 ; i < arr.length ; i++){
        let element = arr[i];
        let j;
        for(j = i - 1; j >= 0 ; j--){
            if(arr[j] > element){
                arr[j+1] = arr[j];
            }else{
                //arr[j] <= element
                break;
            }
        }
        arr[j+1] = element;
    }
}

let arr = [12,31,25,8,32,17];
InsertionSort(arr)
console.log(arr); //[ 8, 12, 17, 25, 31, 32 ]

let myArray = [3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5];
console.log(InsertionSort(myArray)); // undefined

File: SORTING/INSERTIONSORT/InsertionSort.java
package INSERTIONSORT;

public class InsertionSort{
    
    public static void insertionSort(int[] arr) {
        for(int i = 1 ; i < arr.length ; i++){
            int element = arr[i];
            int j;
            for( j = i - 1 ; j >= 0 ; j-- ){
                if(arr[j] > element){
                    arr[j+1] = arr[j];
                }else{
                    // arr[j] <= element
                    break;
                }
            }
            arr[j+1] = element;
        }
    }
    
    public static void main(String[] args) {
        int arr[] = {12,31,25,8,32,17};
        insertionSort(arr);
        for(int i = 0 ; i < arr.length ; i++){
            System.out.println(arr[i]);
        }
    }
}
